src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -1371,20 +1384,52 @@ private void WriteRewriter()
                             WriteLine(GetVisitFunctionDeclaration(node.Name, isOverride: true));
                             Brace();
                             bool hadField = false;
+
                             foreach (Field field in AllNodeOrNodeListFields(node))
                             {
                                 hadField = true;
                                 WriteNodeVisitCall(field);
                             }
+
                             foreach (Field field in AllTypeFields(node))
                             {
                                 hadField = true;
                                 WriteLine("TypeSymbol? {0} = this.VisitType(node.{1});", ToCamelCase(field.Name), field.Name);
                             }
+
+                            foreach (var field in AllNonTypeSymbolOrNonTypeSymbolListFields(node))
+                            {
+                                hadField = true;
+
+                                if (!IsImmutableArray(field.Type, out string elementType))
+                                {
+                                    switch (field.Type.TrimEnd('?'))
+                                    {
+                                        case "Symbol":
+                                            WriteLine($"{field.Type} {ToCamelCase(field.Name)} = this.VisitSymbol(node.{field.Name});");
+                                            break;
+
+                                        default:
+                                            WriteLine($"{field.Type} {ToCamelCase(field.Name)} = this.Visit{field.Type.TrimEnd('?')}(node.{field.Name});");
```

Comment:
Why is this switch necessary? Wouldn't this case work for the "Symbol" as well?

Reply:
> Why is this switch necessary? Wouldn't this case work for the "Symbol" as well?

I guess, given the current naming scheme it would. I started with having a different name for the ```VisitSymbol``` helper, that name didn't fit the pattern. I'll unify.